bazel/external/quiche.BUILD

```diff
@@ -5281,3 +5282,233 @@ envoy_quiche_platform_impl_cc_library(
 )
+
+envoy_cc_library(
+    name = "balsa_enums_lib",
+    srcs = ["quiche/common/balsa/balsa_enums.cc"],
+    hdrs = ["quiche/common/balsa/balsa_enums.h"],
+    copts = quiche_copts,
+    repository = "@envoy",
+    tags = ["nofips"],
+    visibility = ["//visibility:public"],
+    deps = [":quiche_common_platform_export"],
+)
+
+envoy_cc_library(
+    name = "balsa_visitor_interface_lib",
+    hdrs = ["quiche/common/balsa/balsa_visitor_interface.h"],
+    copts = quiche_copts,
+    repository = "@envoy",
+    tags = ["nofips"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":balsa_enums_lib",
+        ":quiche_common_platform_export",
+    ],
+)
+
+envoy_cc_library(
+    name = "noop_balsa_visitor_lib",
+    hdrs = ["quiche/common/balsa/noop_balsa_visitor.h"],
+    copts = quiche_copts,
+    repository = "@envoy",
+    tags = ["nofips"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":balsa_visitor_interface_lib",
+        ":quiche_common_platform_export",
+    ],
+)
+
+envoy_cc_library(
+    name = "standard_header_map_lib",
+    srcs = ["quiche/common/balsa/standard_header_map.cc"],
+    hdrs = ["quiche/common/balsa/standard_header_map.h"],
+    copts = quiche_copts,
+    repository = "@envoy",
+    tags = ["nofips"],
+    deps = [
+        ":quiche_common_text_utils_lib",
+        "@com_google_absl//absl/container:flat_hash_set",
+        "@com_google_absl//absl/strings",
+    ],
+)
+
+envoy_cc_library(
+    name = "framer_interface_lib",
+    hdrs = ["quiche/common/balsa/framer_interface.h"],
+    copts = quiche_copts,
+    repository = "@envoy",
+    tags = ["nofips"],
+    deps = [":quiche_common_platform_export"],
+)
+
+envoy_cc_library(
+    name = "http_validation_policy_lib",
+    srcs = ["quiche/common/balsa/http_validation_policy.cc"],
+    hdrs = ["quiche/common/balsa/http_validation_policy.h"],
+    copts = quiche_copts,
+    repository = "@envoy",
+    tags = ["nofips"],
+    deps = [
+        ":quiche_common_platform_export",
+        ":quiche_common_platform_logging",
+    ],
+)
+
+envoy_cc_library(
+    name = "header_api_lib",
+    hdrs = ["quiche/common/balsa/header_api.h"],
+    copts = quiche_copts,
+    repository = "@envoy",
+    tags = ["nofips"],
+    deps = [
+        ":quiche_common_platform_export",
+        ":quiche_common_platform_lower_case_string",
+        "@com_google_absl//absl/strings",
+    ],
+)
+
+envoy_cc_library(
+    name = "simple_buffer_lib",
+    srcs = ["quiche/common/balsa/simple_buffer.cc"],
+    hdrs = ["quiche/common/balsa/simple_buffer.h"],
+    copts = quiche_copts,
+    repository = "@envoy",
+    tags = ["nofips"],
+    deps = [
+        ":quiche_common_platform",
+        ":quiche_common_platform_bug_tracker",
+        ":quiche_common_platform_export",
+        ":quiche_common_platform_logging",
+        "@com_google_absl//absl/strings",
+    ],
+)
+
+envoy_cc_test(
+    name = "simple_buffer_test",
```

Comment:
Summing up our brief in-person discussion: while not every test is added to CI in order to economize on resources, Balsa tests are different from other QUICHE tests in the sense that they are not run on Chromium.  (Chromium has every single QUICHE test as part of presubmit for every CL, except it does not use Balsa at all, and there are no plans for it.)  Since there are some things common in Envoy and Chromium in which they differ from Google's internal code (for example, char is signed, which can cause subtle changes in behavior, see https://github.com/google/quiche/commit/cd0fcd0f01f34ccf5a6b2ac0c560df56b2f0482e), there is value in adding these to CI on Envoy.